src/goose/statsd/statsd.clj

```diff
@@ -0,0 +1,78 @@
+(ns goose.statsd.statsd
+  (:require
+    [goose.defaults :as d]
+    [goose.heartbeat :as heartbeat]
+    [goose.redis :as r]
+    [goose.utils :as u]
+
+    [clj-statsd :as statsd]))
+
+(defonce jobs-count "jobs.count")
+(defonce jobs-success "jobs.success")
+(defonce jobs-failure "jobs.failure")
+
+(defonce execution-time "job.execution_time")
+
+(defonce retry-latency "retry.latency")
+(defonce schedule-latency "scheduled.latency")
+(defonce execution-latency "execution.latency")
+
+(defonce queue-size "queue.size")
+(defonce schedule-queue-size "scheduled-queue.size")
+(defonce dead-queue-size "dead-queue.size")
+
+(defonce default-opts
+         {:host        "127.0.0.1"
+          :port        8125
+          :sample-rate 1.0
+          :tags        {}})
+
+(defn- build-tags
+  [tags]
+  (map
+    (fn [[key value]] (str (name key) ":" (name value)))
+    tags))
+
+(defn initialize
+  [{:keys [host port]}]
+  (when (and host port)
+    (statsd/setup host port :prefix d/statsd-prefix)))
+
+(defn run-job-and-send-metrics
+  [{:keys [sample-rate tags]} [key latency] fn-name run-fn]
```

Comment:
`fn-name` can be constructed in the `let` here instead of passing in separately?

Reply:
done.